src/tools/crawling_tools.py

```
@@ -172,6 +187,16 @@ async def scrape(ctx: Context, url: str) -> str:
             return json.dumps({"success": False, "url": url, "error": result.error_message}, indent=2)
     except Exception as e:
         return json.dumps({"success": False, "url": url, "error": str(e)}, indent=2)
+    
+    finally:
+        # Always unregister the operation to prevent cleanup blocks
+        try:
+            if 'lifespan_ctx' in locals() and hasattr(lifespan_ctx, 'active_operations'):
+                async with lifespan_ctx.cleanup_lock:
+                    lifespan_ctx.active_operations.discard(operation_id)
+                    logger.info(f"🔓 Unregistered operation {operation_id}, remaining: {len(lifespan_ctx.active_operations)}")
+        except Exception as cleanup_error:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)